cpp/ScanChain_Hgg.C

```diff
@@ -327,6 +325,38 @@ int ScanChain_Hgg(TChain *ch, double genEventSumw, TString year, TString process
       float weight = 1.0;
       if ( isMC ) {
         weight = nt.genWeight();
+      }
+      n_gen_matched_jets = 0;
+      n_gen_matched_in_dijet = 0;
+      LeadPhoton_genPartFlav = 0;
+      SubleadPhoton_genPartFlav = 0;
+      n_gen_matched_jets = 0;
+      n_gen_matched_in_dijet = 0;
+      dijet_lead_gen_match=false;
+      dijet_sublead_gen_match=false;
+      GenHiggs_pt=-999;
+      GenHiggs_eta=-999;
+      GenHiggs_phi=-999;
+      GenHiggs_mass=-999;
+      GenHiggs_dR=-999;
+      GenY_pt=-999;
+      GenY_eta=-999;
+      GenY_phi=-999;
+      GenY_mass=-999;
+      GenY_dR=-999;
+      GenX_pt=-999;
+      GenX_eta=-999;
+      GenX_phi=-999;
+      GenX_mass=-999;
+      GenX_dR=-999;
+      GenBFromHiggs_1_pt=-999;
+      GenBFromHiggs_1_eta=-999;
+      GenBFromHiggs_1_phi=-999;
+      GenBFromHiggs_1_mass=-999;
+      GenBFromHiggs_2_pt=-999; 
+      GenBFromHiggs_2_eta=-999;
+      GenBFromHiggs_2_phi=-999;
+      GenBFromHiggs_2_mass=-999;
```

Comment:
This is inside the event loop to reset every variable into default value, when looping over a new event. L173-L180 is the default definition of each variable, only calls once.

Reply:
And why do you do this only for the gen variables but not the other ones? What am i missing?

Reply:
Hmm... I was looking into this again and I think I now see a reason why these variables (and also others) need to be initialized per event: If they are filled for an event and then not filled for the next, the values of the previous event will have been kept and added to the event, while that one should have had the default values. One step further, it would generally be good practice to initialize all of the variables at the start of the loop, just to avoid any potential misassignments.

As a result, I would like to ask you to fix this in the following way:
- Remove the values from [these lines](https://github.com/cmstas/XToYggHbb_looper/blob/c4c8a771fce6c5ef396857583e41aec315e044b8/cpp/ScanChain_Hgg.C#L173-L180). Just initialize the variables with no numerical values, so that we don't have the illusion that we reset them to those values - we don't.
- Initialize all of those variables [here](https://github.com/cmstas/XToYggHbb_looper/blob/c4c8a771fce6c5ef396857583e41aec315e044b8/cpp/ScanChain_Hgg.C#L325) - this way we make sure that we have the default values at the beginning of each event.

Sorry for the back and forth, it took me some time to understand the real problem...

Reply:
I've put every gen variable initialization in the event loop to avoid the misalignments. Thanks for the suggestions! :)